backend/methods/relative_superiority.py

```diff
@@ -8,43 +8,47 @@ def relative_superiority(m_votes, v_const_seats, v_party_seats,
     """Apportion by Þorkell Helgason's Relative Superiority method"""
 
     m_allocations = deepcopy(m_prior_allocations)
-    num_preallocated_seats = sum([sum(x) for x in m_allocations])
+    num_allocated = sum([sum(x) for x in m_allocations])
     num_total_seats = sum(v_const_seats)
-    for n in range(num_total_seats-num_preallocated_seats):
-        m_votes = threshold_elimination_constituencies(m_votes, 0.0, v_party_seats, m_allocations)
+    for n in range(num_total_seats-num_allocated):
+        m_votes = threshold_elimination_constituencies(m_votes, 0.0,
+                    v_party_seats, m_allocations)
         superiority = []
-        firstin = []
+        first_in = []
         for j in range(len(m_votes)):
             seats_left = v_const_seats[j] - sum(m_allocations[j])
             if not seats_left:
                 superiority.append(0)
-                firstin.append(0)
+                first_in.append(0)
                 continue
 
             next_alloc_num = sum(m_allocations[j]) + 1
-            app_next = apportion1d(m_votes[j], next_alloc_num,
+            alloc_next, div_next = apportion1d(m_votes[j], next_alloc_num,
                                    m_allocations[j], divisor_gen)
-            change = [0 if app_next[0][i] == m_allocations[j][i] else 1
-                      for i in range(len(m_votes[j]))]
-            nextin = change.index(1)
-            new_votes = copy(m_votes[j])
-            new_votes[nextin] = app_next[0][2]
```

Comment:
I guess this was intended as 
`app_next[1][2]`
(makes more sense to update vote counts to divided votes of same list, rather than number of allocated seats of another list)

Although I do not see why the votes shouldn't simply have been set to 0, if the purpose is to make sure the list is not given additional seats while calculating the continuation to find a substitute for comparison.
Setting the votes to the value of the divided votes perhaps serves to make it less likely that the list is skipped over in the continuation, but no guarantee by any means, and the divided votes are not even necessarily lower than the original, as this could have been the first seat allocated to the list.